com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs

```diff
@@ -449,20 +446,27 @@ void UpgradeResourcesIfNeeded()
             m_GlobalSettings.EnsureRuntimeResources(forceReload: true);
             m_GlobalSettings.EnsureEditorResources(forceReload: true);
 
+            bool requiresRayTracingResources = false;
             if (GatherRayTracingSupport(asset.currentPlatformRenderPipelineSettings))
             {
-                m_GlobalSettings.EnsureRayTracingResources(forceReload: true);
+                requiresRayTracingResources = true;
             }
-            else
+            // Also make sure to include ray-tracing resources if at least one of the quality levels needs it
+            else if (rayTracingSupportedBySystem)
             {
-                // If ray tracing is not enabled we do not want to have ray tracing resources referenced
-                m_GlobalSettings.ClearRayTracingResources();
+                int qualityLevelCount = QualitySettings.names.Length;
+                for (int i = 0; i < qualityLevelCount && !requiresRayTracingResources; ++i)
```

Comment:
Maybe just use break in the if instead of second condition of !requiresRayTracingResources
I think it's a bit easier to read that way.